python/pyspark/ml/param/__init__.py

```diff
@@ -484,8 +484,19 @@ def _copyValues(self, to, extra=None):
         :return: the target instance with param values copied
         """
         paramMap = self._paramMap.copy()
-        if extra is not None:
-            paramMap.update(extra)
+        if isinstance(extra, dict):
+            for param, value in extra.items():
+                if isinstance(param, Param):
+                    paramMap[param] = value
+                elif self.hasParam(param):
+                    # set value by name if vaiid
+                    paramMap[self.getParam(param)] = value
```

Comment:
I don't know if we should support this, as it isn't specified this way (by name) elsewhere. Part of the reason is that it can be ambiguous if you have multiple instances of the same transformer in the pipeline? I'm only surprised it wouldn't already fail, so indeed maybe the validation below is reasonable.

Reply:
It can't be ambiguous in a pipeline: since `stages` is the only parameter that pipeline has, that is the only parameter that could be set this way.  If one did pass stages as extra parameters, it would have the same effect as creating a new pipeline with those stages.  No idea why someone would do that, but no compelling reason to stop them.

Reply:
Hm yeah I don't know if there's a use case for passing parameters to elements of a Pipeline this way. It might work, not sure. Anyway I think argument from consistency probably still stands.